cider/function/FunctionLookupEngine.cpp

```diff
@@ -24,15 +24,24 @@
 
 FunctionLookupEnginePtrMap FunctionLookupEngine::function_lookup_engine_ptr_map_ = {};
 
-const FunctionLookupEngine* FunctionLookupEngine::getInstance(
+std::mutex FunctionLookupEngine::s_mutex_;
+
+FunctionLookupEnginePtr FunctionLookupEngine::getInstance(
     const PlatformType from_platform) {
   if (from_platform == PlatformType::SubstraitPlatform ||
       from_platform == PlatformType::PrestoPlatform) {
-    if (function_lookup_engine_ptr_map_.find(from_platform) ==
+    if (function_lookup_engine_ptr_map_.find(from_platform) !=
```

Comment:
can remove L33-36 since you add lock on L37?

Reply:
This is to avoid locking to improve performance when the single instance has been initialized.